qiskit/pulse/parser.py

```diff
@@ -273,12 +277,12 @@ def generic_visit(self, node):
         raise PulseError('Unsupported node: %s' % node.__class__.__name__)
 
 
-def parse_string_expr(source, partial_binding=False):
+def parse_string_expr(source: str, partial_binding: bool = False):
     """Safe parsing of string expression.
 
     Args:
-        source (str): String expression to parse.
-        partial_binding (bool): Allow partial bind of parameters.
+        source: String expression to parse.
+        partial_binding: Allow partial bind of parameters.
 
```

Comment:
Since `partial_binding` option is currently not used and perhaps difficult to realize how it can be used, having an example code in this place is better for developers who may use this in future.

```     
    expr = 'P1 + P2 + P3'
    parsed_expr = parse_string_expr(expr, partial_binding=True)

    # create new PulseExpression
    bound_two = parsed_expr(P1=1, P2=2)
    # evaluate expression
    value1 = bound_two(P3=3)
    value2 = bound_two(P3=4)
    value3 = bound_two(P3=5)
```

Reply:
Great idea!! I will use this example ^